tests/v2/validation/snapshot/snapshot.go

```diff
@@ -115,49 +120,49 @@ func snapshotRestore(t *testing.T, client *rancher.Client, clusterName string, e
-		restoreRKE1(t, client, initialSnapshotValues, etcdRestore, clusterName, clusterID)
+		restoreSnapshotResp, initialSnapshotValues := snapshotRKE1(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, etcdRestore, isRKE1)
+		restoreRKE1(t, client, initialSnapshotValues, etcdRestore, restoreSnapshotResp, clusterName, clusterID)
 	} else {
-		initialSnapshotValues := snapshotV2Prov(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, etcdRestore, isRKE1)
-		restoreV2Prov(t, client, initialSnapshotValues, etcdRestore, clusterName, clusterID)
+		restoreSnapshotConfig, initialSnapshotValues := snapshotV2Prov(t, client, podTemplate, deployment, clusterName, clusterID, localClusterID, etcdRestore, isRKE1)
+		restoreV2Prov(t, client, initialSnapshotValues, etcdRestore, restoreSnapshotConfig, clusterName, clusterID)
 	}
 
 	logrus.Infof("Deleting created workloads...")
 	err = steveclient.SteveType(DeploymentSteveType).Delete(deploymentResp)
 	require.NoError(t, err)
 
-	err = steveclient.SteveType(ServiceType).Delete(serviceResp)
+	err = steveclient.SteveType(serviceType).Delete(serviceResp)
 	require.NoError(t, err)
 
-	err = steveclient.SteveType(IngressSteveType).Delete(ingressResp)
+	err = steveclient.SteveType(ingressSteveType).Delete(ingressResp)
 	require.NoError(t, err)
 }
 
-func snapshotRKE1(t *testing.T, client *rancher.Client, podTemplate corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID, localClusterID string, etcdRestore *etcdsnapshot.Config, isRKE1 bool) initialSnapshotConfig {
+func snapshotRKE1(t *testing.T, client *rancher.Client, podTemplate corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID, localClusterID string, etcdRestore *etcdsnapshot.Config, isRKE1 bool) (*management.Cluster, initialSnapshotConfig) {
+	if etcdRestore.ReplaceNodes != (etcdsnapshot.ReplaceNodes{}) {
+		scaling.ReplaceRKE1Nodes(t, client, clusterName, etcdRestore.Etcd, etcdRestore.ControlPlane, etcdRestore.Worker)
+	}
+
 	existingSnapshots, err := etcdsnapshot.GetRKE1Snapshots(client, clusterID)
 	require.NoError(t, err)
 
 	err = etcdsnapshot.CreateRKE1Snapshot(client, clusterName)
 	require.NoError(t, err)
 
-	clusterResp, err := client.Management.Cluster.ByID(clusterID)
+	restoreSnapshotResp, err := client.Management.Cluster.ByID(clusterID)
 	require.NoError(t, err)
 
 	podErrors := pods.StatusPods(client, clusterID)
 	assert.Empty(t, podErrors)
 
-	if etcdRestore.ReplaceWorkerNode {
-		scaling.ReplaceRKE1Nodes(t, client, clusterName, false, false, true)
-	}
-
-	initialKubernetesVersion := clusterResp.RancherKubernetesEngineConfig.Version
-	require.Equal(t, initialKubernetesVersion, clusterResp.RancherKubernetesEngineConfig.Version)
+	initialKubernetesVersion := restoreSnapshotResp.RancherKubernetesEngineConfig.Version
+	require.Equal(t, initialKubernetesVersion, restoreSnapshotResp.RancherKubernetesEngineConfig.Version)
```

Comment:
this require is redundant, right? since you've just set it in the var above, it seems unnecessary

Reply:
I can remove it

Reply:
Resolved